doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,1275 @@
+   from spyder.api.config.decorators import on_conf_change
+   from spyder.api.translations import get_translation
+
+   from spyder.api.widgets.mixins import SpyderWidgetMixin
+
+
+   # Localization
+   _ = get_translation("spyder_pomodoro_timer.spyder")
+
+
+.. hint::
+
+   The preset imports are a guide to what we will need in our plugin. The ``on_conf_change`` decorator will allow us to propagate the changes in the configuration. ``get_translation`` help us to generate the translation strings and ``SpyderWidgetMixin`` adds to any Widget the attributes and methods needed to integrate with Spyder (icon, style, translation, actions and extra options).
+
+After some research in the Spyder source code we can find that in Spyder there are two types of predefined components for the status bar:
+
+* ``StatusBarWidget``, a class derived from ``QWidget`` and ``SpyderWidgetMixin``, which contains an icon, a label and a spinner (to show the plugin loading).
+* ``BaseTimerStatus``, a class derived from ``StatusBarWidget`` with an internal ``QTimer`` to periodically update its content.
+
+.. note::
+
+   Below, we will be indicating links in github with the diffs between the tags, this as an aid to check the progressive changes that will be made in the code.
+
+`v0.0.1dev-v0.0.2dev widgets.py diff`_
+
+.. _v0.0.1dev-v0.0.2dev widgets.py diff: https://github.com/map0logo/spyder-pomodoro-timer/compare/v0.0.1-dev0...v0.0.2-dev0#diff-f156d182eb57c9fa20d1ca4b9677d13b158447c04f90724d936bb77a50df31bc
+
+Since we want a widget that is periodically updated showing the pomodoro countdown, we will use a ``BaseTimerStatus`` instance.
+
+So, we can safely substitute
+
+.. code-block:: python
+
+   from spyder.api.widgets.mixins import SpyderWidgetMixin
+
+with
+
+.. code-block:: python
+
+   from spyder.api.widgets.status import BaseTimerStatus
+   from spyder.utils.icon_manager import ima
+
+Add an initial import:
+
+.. code-block:: python
+
+   # Third party imports
+   import qtawesome as qta
+
+And add at the end of the file:
+
+.. code-block:: python
+
+   class PomodoroTimerStatus(BaseTimerStatus):
+       """Status bar widget to display the pomodoro timer"""
+
+       ID = "pomodoro_timer_status"
+       CONF_SECTION = "spyder_pomodoro_timer"
+
+       def __init__(self, parent):
+           super().__init__(parent)
+           self.value = "25:00"
+
+       def get_tooltip(self):
+           """Override api method."""
+           return "I am the Pomodoro timer!"
+
+       def get_icon(self):
+           return qta.icon("mdi.av-timer", color=ima.MAIN_FG_COLOR)
+
+.. hint::
+
+    Spyder needs ``ID`` and ``CONF_SECTION`` to be defined, the constructor calls the parent class constructor and initializes the label ``value``.
+
+We add a tooltip to verify the existence of our plugin. Since Spyder uses ``qtawesome`` (is another of our projects that eases the incorporation of iconic fonts into Qt GUIs) we can select our icon using ``qta-browser`` from the terminal.
```

Comment:
```suggestion
We add a tooltip to verify the existence of our plugin. Since Spyder uses ``qtawesome`` (another of our projects that eases the incorporation of iconic fonts into Qt GUIs) we can select our icon using ``qta-browser`` from the terminal.
```